### DISCUSSION TITLE
Facing Performance Degradation After Moving To New Architecture

### DISCUSSION THREAD

**Discussion by udasi-jew-peter** [reactwg/react-native-new-architecture#207] [Q&A]
# Context

So we have an app that was on RN 0.72.8 and we upgraded to 0.74.3 with new architecture enabled. We saw severe performance degradation upon doing that.

We have made a reproducer for this:
- RN 0.72.8 - https://github.com/udasi-jew-peter/new-arch-perf-degradation/tree/feat/rn-72
- RN 0.74.3 with New Arch - https://github.com/udasi-jew-peter/new-arch-perf-degradation/tree/feat/rn-74-new-architecture

Attaching some screenshots of profiling data. The first profile is of 0.74.3 with New Architecture. The other 2 are of 0.72.8.

<img width="1728" alt="Screenshot 2024-07-30 at 4 37 04 PM" src="https://github.com/user-attachments/assets/a41ce5a0-da15-417e-989c-f83607b3a025">
<img width="1728" alt="Screenshot 2024-07-30 at 4 38 27 PM" src="https://github.com/user-attachments/assets/05cda225-63a9-46af-bf9a-93eb40c00c3a">

**Comment by cortinico** [comment]
Can we avoid mixing version bump + New Architecture switch?
Do you have a benchmark for 0.74 **without** NewArch?

Also I see your reproducer has a bunch of dependencies included:

```
    "@react-navigation/bottom-tabs": "6.5.11",
    "@react-navigation/native": "6.1.9",
    "@react-navigation/native-stack": "6.9.17",
    "@shopify/restyle": "2.0.0",
    "moti": "0.29",
    "react": "18.2.0",
    "react-native": "0.74.3",
    "react-native-gesture-handler": "2.16.2",
    "react-native-intersection-observer": "0.0.8",
    "react-native-reanimated": "3.14.0",
    "react-native-safe-area-context": "4.10.5"
```

Have you bisected them to see which one is introducing the regression you're seeing?
